tests/LinAlg/matrixTestsSparse.hpp

```diff
@@ -732,6 +732,74 @@ class MatrixTestsSparse : public TestBase
     return fail;
   }
 
+  /**
+  * @brief Copy 'n_rows' rows from matrix 'A', started from 'A_rows_st', to the rows started from 'B_rows_st' in 'B'.
+  * The non-zero elements start from 'B_nnz_st' will be replaced by the new elements.
+  *
+  * @pre 'A' must have exactly, or more than 'n_rows' rows after row 'A_rows_st'
+  * @pre 'B' must have exactly, or more than 'n_rows' rows after row 'B_rows_st'
+  * @pre 'B_nnz_st' + the number of non-zeros in the copied the rows must be less or equal to B.nnz
+  * @pre User must know the nonzero pattern of A and B. Assume non-zero patterns of A and B wont change, and A is a submatrix of B
+  * @pre Otherwise, this function may replace the non-zero values and nonzero patterns for the undesired elements.
+  */
+  int
+  copyRowsBlockFrom(
+    hiop::hiopMatrixSparse& A,
+    hiop::hiopMatrixSparse& B,
+    local_ordinal_type A_rows_st, local_ordinal_type n_rows,
+    local_ordinal_type B_rows_st, local_ordinal_type B_nnz_st
+    )
+  {
+    const local_ordinal_type* A_iRow = getRowIndices(&A);
+    const local_ordinal_type* A_jCol = getColumnIndices(&A);
+    const local_ordinal_type A_nnz = A.numberOfNonzeros();
+
+    const local_ordinal_type* B_iRow = getRowIndices(&B);
+    const local_ordinal_type* B_jCol = getColumnIndices(&B);
+    const local_ordinal_type B_nnz = B.numberOfNonzeros();
+
+    auto nnz_A_need_to_copy{0};
+    for(auto k=0;k<A_nnz;k++){
+      if(A_iRow[k] >= A_rows_st && A_iRow[k] < A_rows_st + n_rows )
+      {
+        nnz_A_need_to_copy++;
+      }
+      // assume matrix element is ordered by row
```

Comment:
```suggestion
    local_ordinal_type nnz_A_need_to_copy = 0;
    for(local_ordinal_type k=0; k<A_nnz; ++k){
      if(A_iRow[k] >= A_rows_st && A_iRow[k] < A_rows_st + n_rows )
      {
        nnz_A_need_to_copy++;
      }
      // assume matrix element is ordered by row
```
This should fix #110.